app/server_inactivity.go

```diff
@@ -112,6 +112,7 @@ func (s *Server) takeInactivityAction() {
 	}
 
 	for _, user := range users {
+		user := user
```

Comment:
Do you know why this actually fixes it?

Doesn't seem like it changes anything.

Reply:
Yes, it does fix it.

Reply:
Yeah but _why_ does this work, when not having it doesn't? 

I'm assuming there's a race between the go routines being launched and the value changing for `user` but I'm concerned that this is just masking the problem, since all this does is re sets the user value. 

I think the proper way to do this is editing the actual go routine to look like this:

```
s.Go(func(user *User) {
			if err := s.EmailService.SendLicenseInactivityEmail(user.Email, name, user.Locale, siteURL); err != nil {
				mlog.Error("Error while sending inactivity reminder email.", mlog.String("user_email", user.Email), mlog.Err(err))
			}
		}(user))
```

So that when the go routine is invoked, the user is passed as an argument to it

Reply:
There’s two ways to solve that problem of goroutines and closures. The one you’re suggesting and the one I have added.

Reply:
The comment with link to the doc was helpful.
![image](https://user-images.githubusercontent.com/12176405/156822121-cfa948f2-554f-4638-847b-8e26ffecde5c.png)
You can see that the doc mentions my solution as well as yours, I've never seen your approach before so it just felt off to me. I'm cool with it though, thanks for clarifying!

Reply:
I ran into this after Allan pointed me to the issue.

> I've never seen your approach before so it just felt off to me. 

This is quite idiomatic and in fact, used in a lot of places in our codebase. See https://github.com/mattermost/mattermost-server/pull/13969.